instagram_scraper.py

```diff
@@ -17,88 +18,132 @@
+
+    yield from send_scrape_request(url, total_count=total_count, existing=existing, short_circuit=short_circuit)
 
-def main(tags, total_count, should_continue):
-    def _single_tag_processing(tag, total_count, existing_links, start):
-        os.makedirs(f'data/{tag}', exist_ok=True)
-        with open(f'data/{tag}/data.csv', 'a' if existing_links else 'w', newline='', encoding='utf-8') as csvfile:
+
+def main(tags: List[str], users: List[str], total_count: int=50, should_continue: bool=False):
+    """
+    :param tags:
+        List of tags to be scraped
+    :param users:
+        List of users to be scraped
+    :param total_count:
+        total number of images to be scraped
+    :param should_continue
+        Flag for whether or not we should read from disk and skip existing URLs
+        
+    Scrapes user and hashtag images from Instagram
+    """
+    def _single_input_processing(target: str, total_count: int, existing_links: set, start: int, mode: str='tag'):
+        os.makedirs(f'data/{target}', exist_ok=True)
+        with open(f'data/{target}/data.csv', 'a' if existing_links else 'w', newline='', encoding='utf-8') as csvfile:
             writer = csv.writer(csvfile, delimiter=',')
-            for count, (url, caption, hashtags, mentions) in enumerate(scrape_instagram_tag(
-                tag, total_count, existing_links), start):
+            for count, (url, caption, hashtags, mentions) in enumerate(scrape_instagram(
+                target, total_count, existing_links, mode), start):
 
                 try:
                     req = requests.get(url)
-                    with open(f'data/{tag}/{count}.jpg', 'wb') as img:
+                    with open(f'data/{target}/{count}.jpg', 'wb') as img:
                         img.write(req.content)
                 except:
                     print(f'An error occured while downloading {url}')
                 else:
+                    file_index = count + 1
                     writer.writerow([
-                        f'{count}.jpg',
+                        f'{file_index}.jpg',
                         url,
                         caption.replace('\n', '\\n'),
                         ', '.join(hashtags),
                         ', '.join(mentions)
                     ])
-                    print(f'[{tag}] downloaded {url} as {count}.jpg in data/{tag}')
-
-    for tag in tags:
-        existing_links = set()
-        start = 0
-        if os.path.exists(f'data/{tag}/data.csv') and should_continue:
-            with open(f'data/{tag}/data.csv', newline='', encoding='utf-8') as csvfile:
-                reader = csv.reader(csvfile)
-                for i, row in enumerate(reader):
-                    existing_links.add(row[1])
-                start = i + 1
-        _single_tag_processing(tag, total_count, existing_links, start)
+                    print(f'[{target}] downloaded {url} as {file_index}.jpg in data/{target}')
```

Comment:
This becomes incorrect, since we are downloading as `f'{count}.jpg'` which is one less than `file_index`. Replace `count` with `file_index`, better variable name.